src/oneD/StFlow.cpp

```diff
@@ -451,22 +456,39 @@ void StFlow::evalResidual(double* x, double* rsd, int* diag,
             //      - sum_k(J_k c_p_k / M_k) dT/dz
             //-----------------------------------------------
             if (m_do_energy[j]) {
-                setGas(x,j);
 
-                // heat release term
-                const vector_fp& h_RT = m_thermo->enthalpy_RT_ref();
-                const vector_fp& cp_R = m_thermo->cp_R_ref();
+                setGas(x,j);
+                double dtdzj = dTdz(x,j);
                 double sum = 0.0;
                 double sum2 = 0.0;
-                for (size_t k = 0; k < m_nsp; k++) {
-                    double flxk = 0.5*(m_flux(k,j-1) + m_flux(k,j));
-                    sum += wdot(k,j)*h_RT[k];
-                    sum2 += flxk*cp_R[k]/m_wt[k];
+                
+                if(m_thermo->type() == "IdealGas")
+                {
+                    vector_fp cp_R(m_nsp, 0.0), h_RT(m_nsp, 0.0);
+                    m_thermo->getCp_R(&cp_R[0]);
```

Comment:
Here and elsewhere, please make sure your code follows the existing style that we fairly uniformly use, as below.
```suggestion
                if (m_thermo->type() == "IdealGas") {
                    vector_fp cp_R(m_nsp, 0.0), h_RT(m_nsp, 0.0);
                    m_thermo->getCp_R(&cp_R[0]);
```

Reply:
Corrected.